compiler/rustc_parse/src/lexer/mod.rs

```diff
@@ -158,172 +327,6 @@ impl<'a> StringReader<'a> {
         }
     }
 
-    /// Turns simple `rustc_lexer::TokenKind` enum into a rich
-    /// `rustc_ast::TokenKind`. This turns strings into interned
-    /// symbols and runs additional validation.
-    fn cook_lexer_token(&self, token: rustc_lexer::TokenKind, start: BytePos) -> Option<TokenKind> {
```

Comment:
Can we get the same win by slapping `#[inline(always)]` here? I think it's beneficial for readability to move this chunk of boilerplate out of the ws-skipping loop.

Reply:
Alas, no. It was always getting inlined by the compiler, but it's faster when inlined by hand. I guess the compiler can't do as good a job with the `Option` fiddling.